### PR TITLE
[BEAM-3137] Improved validation of DynamicDestinations results in BigQueryIO.write

### PR DESCRIPTION
**Pull request by jkff**
https://issues.apache.org/jira/browse/BEAM-3137

R: @reuvenlax

### CODE REVIEW

**Comment by jkff** [comment]
Fixed checkstyle, PTAL.

**Comment by jkff** [comment]
Reuven is traveling and busy, so
R: @chamikaramj

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteTables.java [hunk 1]
```diff
@@ -106,17 +108,29 @@ public void processElement(ProcessContext c) throws Exception {
...
+        tableSchema =
+            BigQueryHelpers.fromJsonString(jsonSchemas.get(destination), TableSchema.class);
       } else {
         tableSchema = dynamicDestinations.getSchema(destination);
-        if (tableSchema != null) {
-          jsonSchemas.put(destination, BigQueryHelpers.toJsonString(tableSchema));
-        }
+        checkArgument(
+            tableSchema != null,
+            "DynamicDestinations.getSchema() may not return null, "
```
Comment (chamikaramj) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteTables.java]: Here also, we should probably mention that DynamicDestinations.getSchema() may return null if firstPaneCreateDisposition == CreateDisposition.CREATE_NEVER.

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTest.java [hunk 2]
```diff
@@ -1251,6 +1250,32 @@ public void testWriteFailedJobs() throws Exception {
...
+        .withJobService(new FakeJobService())
+        .withDatasetService(new FakeDatasetService());
+
+    Pipeline p = TestPipeline.create(bqOptions);
+
+    PCollectionView<Map<String, String>> view =
+        p.apply("Create schema view", Create.of(KV.of("foo", "bar"), KV.of("bar", "boo")))
+            .apply(View.<String, String>asMap());
+    p.apply(Create.empty(TableRowJsonCoder.of()))
+        .apply(BigQueryIO.writeTableRows().to("dataset-id.table-id")
```
Comment (chamikaramj) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTest.java]: Also add a test for DynamicDestinations.getTable().

**Comment by jkff** [comment]
All done, thanks!

**Pull request closed without merging**